### PR TITLE
finished working on file strformat_lab.py

### PR DESCRIPTION
**Pull request by ldenney89**
Pull request for String Format Lab Assignment Submission. Also for mailroom.py submission.

assignment file:
strformat_lab.py
mailroom.py

### CODE REVIEW

**Comment by ldenney89** [comment]
Also submitting mailroom.py for review.

File: students/laura_denney/lesson03/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+# Dev:   LDenney
+# Date:  October 11, 2018
+# ChangeLog: (Who, When, What)
+#   Laura Denney, 10/11/18, Created File
+#   Laura Denney, 10/12/18, Modified File
+#-------------------------------------------------#
+
+tabl = [["john doe", 100, 200], ["laura denney", 5],
+        ["bill gates", 4000], ["samuel jackson", 1, 2, 3],
+        ["mr. bean", 500, 100]]
```
Comment (natasha-aleksandrova) [inline students/laura_denney/lesson03/mailroom.py]: 👍

File: students/laura_denney/lesson03/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        if response == "1":
+            send_thank_you()
+        elif response == "2":
+            create_report()
+        elif response == "3":
+            continue
+        else:
+            print("\nThat is not a valid selection. Please choose 1, 2 or 3.")
+    else:
+        print("\nYou have chosen to quit. Have a good day!")
```
Comment (natasha-aleksandrova) [inline students/laura_denney/lesson03/mailroom.py]: nice work on main entry flow

File: students/laura_denney/lesson03/mailroom.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+def print_list():
+    names = []
+    for item in tabl:
+        names += [item[0].title()]
+    strformat = "\nWe have {} current donors: " + ", ".join(["{}"] * len(names))
+    print(strformat.format(len(names), *names))
+
+def check_list(name):
+    for x in tabl:
+        if name in x:
```
Comment (natasha-aleksandrova) [inline students/laura_denney/lesson03/mailroom.py]: prefer to use good readable variable names like: `for donor in table`

File: students/laura_denney/lesson03/mailroom.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    header = '''
+Donor Name                | Total Given | Num Gifts | Average Gift
+------------------------------------------------------------------'''
+    print(header)
+    strformat = '{:<26}${:>13}{:^12}${:>13}'
+    for row in tabl:
+        sumamount = sum(row[1:])
+        numberdonations = len(row[1:])
+        average = sumamount/numberdonations
+        print(strformat.format(row[0].title(),round(sumamount,2), numberdonations, round(average,2 )))
```
Comment (natasha-aleksandrova) [inline students/laura_denney/lesson03/mailroom.py]: please check requirements, the report should be sorted

**Comment by ldenney89** [comment]
Hi Natasha,

I think I completely missed the part where it said it should be sorted, I
was so excited I got the formatting.

As I've already submitted the assignment for assessment, is it to late to
make any changes?

-Laura Denney

**Comment by natasha-aleksandrova** [comment]
no worries! we will continue working on the mailroom program by adding features/revisions in later lessons. 

Go ahead and apply the sorting feature in the next revision 🙂

**Comment by ldenney89** [comment]
Awesome, sounds like a plan. Thanks for the response!

**Pull request merged**